eclass/cmake-utils.eclass

```diff
@@ -652,25 +652,57 @@ enable_cmake-utils_src_compile() {
 }
 
 _ninjaopts_from_makeopts() {
-	if [[ ${NINJAOPTS+set} == set ]]; then
+	[[ ${NINJAOPTS+set} == set ]] && \
 		return 0
-	fi
-	local ninjaopts=()
+	eshopts_push -s extglob
+	NINJAOPTS=
+	local jobs= keep= load=
 	set -- ${MAKEOPTS}
 	while (( $# )); do
-		case $1 in
-			-j|-l|-k)
-				ninjaopts+=( $1 $2 )
-				shift 2
-				;;
-			-j*|-l*|-k*)
-				ninjaopts+=( $1 )
-				shift 1
-				;;
-			*) shift ;;
+		case "$1" in
+			-j|--jobs)
+				if [ -n "$2" ] && [[ "$2" =~ ^[0-9]+$ ]] ;then
+					jobs=$2
+					shift
+				else
+					# `man 1 make`:
+					# 	If the -j option is given without an argument, make will not limit
+					# 	the number of jobs that can run simultaneously.
+					jobs=99
+				fi
+			;;
+			-l|--load-average)
+				if [ -n "$2" ] && [[ "$2" =~ ^[0-9]+\.?[0-9]*$ ]] ;then
+					load="${2%%.*}"
+					shift
+				else
+					# `man 1 make`:
+					#	With no argument, removes a previous load limit.
+					load=
+				fi
+			;;
+			-j*|--jobs=*|-l*|--load-average=*)
+				local arg="${1##*([^0-9])}"
+				case "${1##*(-)}" in
+					j*) jobs=$arg ;;
+					l*) load="${arg%%.*}" ;;
+				esac
+			;;
+			-k|--keep-going)
+				# `man 1 make`:
+				# 	Continue as much as possible after an error
+				# `ninja --help`:
+				# 	keep going until N jobs fail
+				keep=99
```

Comment:
https://github.com/ninja-build/ninja/blob/master/src/ninja.cc#L1027

If you think it needs to be in the `--help` output, create a patch for ninja instead of being stupid in our eclass.